src/main/java/com/bisnode/opa/client/rest/OpaRestClient.java

```diff
@@ -31,9 +32,12 @@ public OpaRestClient(OpaConfiguration opaConfiguration, HttpClient httpClient, O
      * Create {@link java.net.http.HttpRequest.Builder} with configured url using provided endpoint
      *
      * @param endpoint desired opa endpoint
+     * @throws OpaClientException if URL or endpoint is invalid
      */
     public HttpRequest.Builder getBasicRequestBuilder(String endpoint) {
-        return HttpRequest.newBuilder(URI.create(opaConfiguration.getUrl() + endpoint));
+        String url = opaConfiguration.getUrl() + "/" + Optional.ofNullable(endpoint).orElseThrow(() -> new OpaClientException("Invalid endpoint: " + endpoint));
```

Comment:
A suggestion: maybe we should throw `MalformedUrlException` or `InvalidEndpointException` that is derived from `OpaClientException`? So the user can react accordingly.